src/twisted/internet/test/test_base.py

```diff
@@ -155,6 +158,38 @@ def fakeGetHostByName(name):
         result.put(IOError("The I/O was errorful"))
 
 
+    def test_resolverGivenStr(self):
+        """
+        L{ThreadedResolver.getHostByName} is passed L{str}.
+        """
+        calls = []
+
+        @implementer(IResolverSimple)
+        class FakeResolver(object):
+            def getHostByName(self, name, timeouts=()):
+                calls.append(name)
+                return Deferred()
+
+        class JustEnoughReactor(ReactorBase):
+            def installWaker(self):
+                pass
+
+        fake = FakeResolver()
+        reactor = JustEnoughReactor()
+        reactor.installResolver(fake)
+        rec = FirstOneWins(Deferred())
+        reactor.nameResolver.resolveHostName(
+            rec, u"example.invalid")
```

Comment:
So, @glyph and I originally decided "nope, just make them pass idna". Turns out, Agent will DE-IDNA IT, and then SEND UNICODE TO THE API. So, I kinda have to now. Buh bow, Hawkie loses.